src/front/glsl/parser.rs

```diff
@@ -245,12 +245,10 @@ pomelo! {
         match fc.kind {
             FunctionCallKind::TypeConstructor(ty) => {
                 let h = if fc.args.len() == 1 {
-                    let kind = match extra.module.types[ty].inner {
-                        TypeInner::Scalar{kind, ..} => kind,
-                        TypeInner::Vector{kind, ..} => kind,
-                        _ => {
-                            return Err(ErrorKind::SemanticError("Can only cast to scalar or vector"));
-                        }
+                    let kind = if let Some(kind) = extra.module.types[ty].inner.scalar_kind() {
```

Comment:
this could be written simpler as
```rust
let kind = extra.module.types[ty].inner
    .scalar_kind()
    .ok_or(ErrorKind::SemanticError("Can only cast to scalar or vector"))?;
```

Reply:
Damn, when do I learn to remember those nice FP helper functions Rust has 😃